pydantic/_internal/_generate_schema.py

```diff
@@ -1741,8 +1741,21 @@ def _pattern_schema(self, pattern_type: Any) -> core_schema.CoreSchema:
             raise PydanticSchemaGenerationError(f'Unable to generate pydantic-core schema for {pattern_type!r}.')
 
     def _hashable_schema(self) -> core_schema.CoreSchema:
+        # json cannot be validated by isinstance, so we manually construct a schema
+        json_schema = core_schema.union_schema(
+            [
+                core_schema.str_schema(),
+                core_schema.int_schema(),
+                core_schema.float_schema(),
+                core_schema.bool_schema(),
+                core_schema.none_schema(),
```

Comment:
What about `tuple`, `frozenset`, or other types like that? There might be a few more we want to include. I suppose we can accept slow validation here given that the type is so broad?